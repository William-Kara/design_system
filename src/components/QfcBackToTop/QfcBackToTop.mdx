import { Meta } from '@storybook/blocks';

<Meta title="Elements/BackToTop" />

# Definition & Usage

A back-to-top button that allows visitors to quickly scroll back to the top of the page with a single click.

It only appears when the user have scroll down.
