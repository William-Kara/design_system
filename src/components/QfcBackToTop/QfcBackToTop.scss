$button-size: 70px;
$laptop-placement: 5rem;
$touch-placement: 2.5rem;

.qfc-back-to-top {
  border-radius: 50%;
  width: $button-size;
  height: $button-size;
  background-color: $yellow;
  box-shadow: 0 3px 12px #000000;
  position: fixed;
  right: $laptop-placement;
  bottom: $laptop-placement;
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 50;

  &:hover {
    cursor: pointer;
  }
  &:active {
    scale: 0.95;
    transition: scale 0.05s;
  }

  .material-icons {
    color: $black;
  }

  @include touch {
    bottom: $touch-placement;
    right: $touch-placement;
  }
}

.v-enter-active,
.v-leave-active, .qfc-back-to-top {
  transition: opacity 0.5s ease;
}

.v-enter-from,
.v-leave-to {
  opacity: 0;
}
