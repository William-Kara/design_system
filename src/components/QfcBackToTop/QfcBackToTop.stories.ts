import BackToTop from './QfcBackToTop.vue';
import type { Meta, StoryObj } from '@storybook/vue3';

const meta: Meta<typeof BackToTop> = {
  component: BackToTop,
  title: 'Elements/BackToTop',
  argTypes: {},
  decorators: [
    () => ({
      template: '<div style="height: 100em;"><h1>Scroll down to see the button</h2> <story/></div>',
    }),
  ],
};
export default meta;

type Story = StoryObj<typeof meta>;

export const Primary: Story = {
  args: {},
};
