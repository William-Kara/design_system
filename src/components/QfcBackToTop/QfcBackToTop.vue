<template>
  <Transition>
    <div v-if="isScrolling" @click="scrollToTop" class="qfc-back-to-top">
      <span class="material-icons">north</span>
    </div>
  </Transition>
</template>

<script lang="ts" setup>
import { useWindowScroll } from '@vueuse/core';
import { computed } from 'vue';

const { y } = useWindowScroll();
const isScrolling = computed(() => y.value > 20);
const scrollToTop = () => window.scrollTo({ top: 0, behavior: 'smooth' });
</script>

<style lang="scss" scoped src="./QfcBackToTop.scss" />
