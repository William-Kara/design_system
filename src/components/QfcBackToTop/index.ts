import QfcBackToTop from './QfcBackToTop.vue';

export { QfcBackToTop };
export default QfcBackToTop;
