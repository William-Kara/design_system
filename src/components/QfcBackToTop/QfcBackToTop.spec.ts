import { it, describe, beforeEach, expect } from 'vitest';

import { shallowMount } from '@vue/test-utils';
import QfcBackToTop from './QfcBackToTop.vue';

let wrapper: any = null;

const propsValues = {};

beforeEach(() => {
  wrapper = shallowMount(QfcBackToTop, {
    propsData: propsValues,
  });
});

describe('QfcBackToTop Component', () => {
  it('should mount properly', () => {
    expect(wrapper).toBeTruthy();
  });

  it('should have correct props', () => {
    expect(wrapper.props()).toEqual(propsValues);
  });

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
