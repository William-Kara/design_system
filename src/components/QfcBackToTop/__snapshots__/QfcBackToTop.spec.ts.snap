// Vitest Snapshot v1

exports[`QfcBackToTop Component > should render correctly 1`] = `
"<transition-stub data-v-1dc72de1=\\"\\" appear=\\"false\\" persisted=\\"false\\" css=\\"true\\">
  <!--v-if-->
</transition-stub>"
`;
